File: 0009-palindrome-number/0009-palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
        int temp=x;
        int sum=0;
        
        while (x>0){
            int remainder=x%10;
            sum=sum*10+remainder;
            x=x/10;
        }
    if(temp==sum)  {
        return  true;
    }else{
        return false;
    }
    }
}

File: 0009-palindrome-number/0009-palindrome-number.c
bool isPalindrome(int n) {
    long long div = 1, rev = 0;
    while (div <= n) {
        long long ld = (n % (div * 10)) / div;
        rev = rev * 10 + ld;
        div = div * 10;
    }
    return rev == n;
}